src/lib/annotations/doc/DocAnnotator.js

```diff
@@ -834,59 +806,39 @@ class DocAnnotator extends Annotator {
         let consumed = false;
         let activeThread = null;
 
-        // Destroy any pending highlights on click outside the highlight
-        const pendingThreads = this.getThreadsWithStates(PENDING_STATES);
```

Comment:
Cleaned it up so we only iterate through the threads on the current page once instead of using getThreadsWithStates()